lib/core/src/persist/mod.rs

```diff
@@ -224,3 +224,87 @@ impl Persister {
         Ok(payments)
     }
 }
+
+#[cfg(test)]
+pub(crate) mod test_utils {
+    use anyhow::{anyhow, Result};
+    use bip39::rand::{self, distributions::Alphanumeric, Rng};
+    use tempdir::TempDir;
+
+    use super::{Network, PaymentTxData, PaymentType, Persister};
+
+    pub(crate) struct TempPersister {
```

Comment:
Why do we need this TempPersister wrapper? Can't we use the Persister itself?

Reply:
The TempPersister exists because we create a `TempDir`, which gets deleted automatically once the tests are over. We need to store this object in memory until we are done with the test, then the directory will be deleted automatically once it goes out of scope.

Reply:
It is another good argument to use in-memory persister.